platform-api/che-core-api-builder/src/main/java/org/eclipse/che/api/builder/internal/BuilderCleaner.java

```diff
@@ -0,0 +1,96 @@
+package org.eclipse.che.api.builder.internal;
+
+import java.io.File;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.eclipse.che.commons.lang.IoUtil;
+import org.eclipse.che.commons.lang.concurrent.ThreadLocalPropagateContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+/**
+ * Manages clean-up operations for build tasks of local slave builders.
+ * 
+ * @author Tareq Sharafy
+ */
+@Singleton
+public class BuilderCleaner {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BuilderCleaner.class);
+
+    private final BuilderRegistry builders;
+    private final long keepResultTimeMillis;
+    private final ScheduledExecutorService cleanupScheduler;
+
+    @Inject
+    public BuilderCleaner(BuilderRegistry builders, @Named(Constants.KEEP_RESULT_TIME) int keepResultTime) {
+        this.builders = builders;
+        this.keepResultTimeMillis = TimeUnit.SECONDS.toMillis(keepResultTime);
+        this.cleanupScheduler = Executors.newSingleThreadScheduledExecutor(
+                new ThreadFactoryBuilder().setNameFormat("BuilderManagerScheduler-%d").setDaemon(true).build());
+    }
+
+    @PostConstruct
```

Comment:
you can use https://github.com/codenvy/che-core/blob/3.x/commons/che-core-commons-schedule/src/main/java/org/eclipse/che/commons/schedule/ScheduleRate.java
docs
https://github.com/eclipse/che/tree/master/core/commons/che-core-commons-schedule